### PR TITLE
fix(client): warn campers if variable is redefined

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

There are a possibility syntax errors campers might not readily solve intefering with the tests. Rather than getting a message like "You should declare a `xpText` variable with `const`." , the challenge stops working on the first text involving a variable (The text one). 
 
Another possibility to solve this issue is moving the xpText and healthText and goldText upwards , but I like my tests grouped and in order. 

Context: https://forum.freecodecamp.org/t/learn-basic-javascript-by-building-a-role-playing-game-step-29/697745/2

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
While this does fix this specific challenge, the underlying problem is that the tests should not even run if there are syntax errors.

In general, except for certain special cases where the learner needs to create an error, I think we should just show them something like this:

> Your code has a syntax error, see the console for details.

In this particular challenge, they'd see

```
SyntaxError: unknown: Identifier 'xp' has already been declared. (13:6)

  11 | const button3 = document.querySelector("#button3");
  12 | const text = document.querySelector("#text");
> 13 | const xp = document.querySelector("#xpText");
     |       ^
  14 | const health = document.querySelector("#healthText");
  15 | const gold = document.querySelector("#goldText");
  16 | const monsterStats = document.querySelector("#monsterStats");
  ```

which is quite helpful.

**Comment by a2937** [comment]
Okay we have an existing syntax error message. Should that error output be modified (and the behavior of our Python code that logs to the console as well as the preview); or would it be better to just use a new error output?

File: client/i18n/locales/english/translations.json [hunk 1]
```diff
@@ -390,7 +390,7 @@
     "running-tests": "// running tests",
     "tests-completed": "// tests completed",
     "console-output": "// console output",
-    "syntax-error": "Your code raised an error before any tests could run. Please fix it and try again.",
+    "syntax-error": "Your code has a syntax error, see the console for details.",
```
Comment (ilenia-magoni) [inline client/i18n/locales/english/translations.json:393]: syntax error is too specific for python, there are many different errors possible (TypeError, ValueError etc). Also it's terminal, not console. Maybe it needs two different strings?

Comment (a2937) [reply client/i18n/locales/english/translations.json]: Got it chief.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This is looking good, @a2937. There are a few issues, though.

If there are multiple failing tests, all failing because of the same syntax error, you get duplicated messages in the console:

image

Also, as you can see from the image, the `Your code has a syntax error, see the console for details.` message appears in the console, but that's not helpful. It's already shown in the lower jaw and that's enough.

Finally, the error not being an instance of `chai.AssertionError` doesn't imply that the user has a syntax error. It just means that it wasn't raised by a chai assertion.

Going back to the image I posted, that was generated by `const locati ons = [];`. When you look at the console before running the tests you see

```
SyntaxError: unknown: Missing initializer in const declaration. (19:12)

  17 | const monsterName = document.querySelector("#monsterName");
  18 | const monsterHealthText = document.querySelector("#monsterHealth");
> 19 | const locati ons = [];
     |             ^
  20 |
  21 | // initialize buttons
  22 | button1.onclick = goStore;
  ```
  
This is what needs to be shown to the learner, not `[ReferenceError: locations is not defined]` which appears because none of the user's code was evaluated and so `locations` doesn't exist.

**Comment by a2937** [comment]
I meant to push days ago but it never actually went out. I don't think I did something quite right though.

**Pull request closed without merging**